### PR TITLE
Add a RAII class for setting and restoring memory tracker

### PR DESCRIPTION
**Pull request by sduzh**

### CODE REVIEW

**Review by dirtysalt — APPROVED** [review]

**Comment by dirtysalt** [comment]
Interesting.

**Review by trueeyu — APPROVED** [review]

File: be/src/exec/pipeline/scan_operator.cpp [hunk 1]
```diff
@@ -104,12 +104,11 @@ void ScanOperator::_trigger_next_scan(RuntimeState* state) {
...
-        DeferOp op([&] {
-            tls_thread_status.set_mem_tracker(prev_tracker);
-            _is_io_task_active.store(false, std::memory_order_release);
-        });
-        _chunk_source->buffer_next_batch_chunks_blocking(_batch_size, _is_finished);
+        {
+            SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(state->instance_mem_tracker());
+            _chunk_source->buffer_next_batch_chunks_blocking(_batch_size, _is_finished);
+        }
+        _is_io_task_active.store(false, std::memory_order_release);
```
Comment (starrocks-xueli) [inline be/src/exec/pipeline/scan_operator.cpp:111]: why not preserve the order before for `_is_io_task_active`?

**Comment by sduzh** [comment]
run starrocks_be_unittest

File: be/src/exec/pipeline/scan_operator.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/exec/pipeline/scan_operator.cpp:111]: The order is the same as before.

**Pull request merged**